src/snowflake/snowpark/session.py

```diff
@@ -451,7 +452,8 @@ def __init__(
         self._conf = self.RuntimeConfig(self, options or {})
         self._tmpdir_handler: Optional[tempfile.TemporaryDirectory] = None
         self._runtime_version_from_requirement: str = None
-
+        # make sure to close the session before exit
+        atexit.register(self.close)
```

Comment:
Does this work in store procedure?

Reply:
It will. Would that be a problem?

Reply:
Actually no, there is no active session added in SP, so it won't close any session in SP.